mclagsyncd/mclaglink.h

```diff
@@ -90,36 +134,121 @@ struct mclag_fdb
 
 };
 
+
+//MCLAG Domain Key 
+struct mclagDomainEntry
+{
+   unsigned int domain_id;
+
+   mclagDomainEntry() {}
+   mclagDomainEntry(unsigned int id):domain_id(id) {}
+
+   bool operator <(const mclagDomainEntry &domain) const
+   {
+       return domain_id < domain.domain_id;
+   }
+
+   bool operator ==(const mclagDomainEntry &domain) const
+   {
+       return (domain_id == domain.domain_id);
+   }
+};
+
+
+//MCLAG Domain Data
+struct mclagDomainData
+{
+    std::string source_ip;
+    std::string peer_ip;
+    std::string peer_link;
+    int keepalive_interval;
+    int session_timeout;
+
+    mclagDomainData()
+    {
+        keepalive_interval = -1;
+        session_timeout    = -1;
+    }
+
+    bool mandatoryFieldsPresent() const
+    {
+        return (!source_ip.empty() && !peer_ip.empty());
+    }
+
+    bool allFieldsEmpty() const
+    {
+        return (source_ip.empty() && peer_ip.empty() && peer_link.empty() && keepalive_interval == -1 && session_timeout == -1);
+    }
+};
+
+typedef  std::tuple<std::string, std::string> vlan_mbr;
+
 class MclagLink : public Selectable {
 public:
     const int MSG_BATCH_SIZE;
+    std::map<std::string, std:: string> *p_learn;
     ProducerStateTable * p_port_tbl;
     ProducerStateTable * p_lag_tbl;
     ProducerStateTable * p_tnl_tbl;
     ProducerStateTable * p_intf_tbl;
     ProducerStateTable *p_fdb_tbl;
     ProducerStateTable *p_acl_table_tbl;
     ProducerStateTable *p_acl_rule_tbl;
+    Table *p_mclag_tbl;
```

Comment:
Can't we make all these DBConnector/Table/ProducerStateTable members as private and initialize on object construction?
E.g.,
https://github.com/Azure/sonic-swss/blob/cba65760393d2380785c407158609f4782777718/cfgmgr/buffermgr.cpp#L18
or
https://github.com/Azure/sonic-swss/blob/cba65760393d2380785c407158609f4782777718/natsyncd/natsync.cpp#L40

Reply:
@akokhan 
thanks for your comments.  will try to address it

Reply:
addressed all review comments. thanks